addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,207 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
             $carriers.filter(':checked').click();
         }
 
-        // Asynchronously retrieve every carrier price
-        _.each($carriers, function (carrierInput, k) {
+        self._getCurrentLocation();
+
+        _.each($carriers, async function (carrierInput, k) {
             self._showLoading($(carrierInput));
-            self._rpc({
+            const result = await self._rpc({
                 route: '/shop/carrier_rate_shipment',
                 params: {
                     'carrier_id': carrierInput.value,
                 },
-            }).then(self._handleCarrierUpdateResultBadge.bind(self));
+            })
+            self._handleCarrierUpdateResult(result)
         });
-
         return this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @private
+     */
+    _getCurrentLocation: async function () {
+        const self=this;
```

Comment:
```suggestion
        const self = this;
```
Note: you actually don't need to define `self` and can just use `this` instead in this function